cmd/kube-controller-manager/app/certificates.go

```diff
@@ -44,7 +44,12 @@ func startCSRSigningController(ctx ControllerContext) (http.Handler, bool, error
 		klog.Warningf("Resource %s is not available now", gvr.String())
 		return nil, false, nil
 	}
-	if ctx.ComponentConfig.CSRSigningController.ClusterSigningCertFile == "" || ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile == "" {
+	hasSingleSigningFile := ctx.ComponentConfig.CSRSigningController.ClusterSigningCertFile == "" || ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile == ""
+	hasKubeletServingFile := ctx.ComponentConfig.CSRSigningController.KubeletServingSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.KubeletServingSignerConfiguration.KeyFile == ""
+	hasKubeletClientFile := ctx.ComponentConfig.CSRSigningController.KubeletClientSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.KubeletClientSignerConfiguration.KeyFile == ""
+	hasKubeAPIServerClientFile := ctx.ComponentConfig.CSRSigningController.KubeAPIServerClientSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.KubeAPIServerClientSignerConfiguration.KeyFile == ""
+	hasLegacyUnknownFile := ctx.ComponentConfig.CSRSigningController.LegacyUnknownSignerConfiguration.CertFile == "" || ctx.ComponentConfig.CSRSigningController.LegacyUnknownSignerConfiguration.KeyFile == ""
```

Comment:
ow, my eyes... maybe

```go
csrConfig := ctx.ComponentConfig.CSRSigningController
hasSingleSigningFile := csrConfig.ClusterSigningCertFile...
...
```
?

Reply:
also, are all of these right? shouldn't these be checking if either of the params are *not* empty?

Reply:
fixed names and updated to use the helper with a correctly named function.